### DISCUSSION TITLE
[AEMaaCS] startup_hook no longer runs after updating to 2024.4.15860.20240410T113939Z

### DISCUSSION THREAD

**Discussion by teropikala** [Netcentric/accesscontroltool#697] [General]
Hello, 

I've noticed in few environments that "startup_hook" no longer runs after I've updated environment to latest version 2024.4.15860.20240410T113939Z 

"startup_hook_image_build" does run correctly and there hasn't been any changes to the way ACTool is installed. 

In the screenshot below there's one successful deployment at the bottom and then 2 deployment attempts where "startup_hook"  wasn't included. 

<img width="1365" alt="Screenshot 2024-04-15 at 14 34 44" src="https://github.com/Netcentric/accesscontroltool/assets/3505319/b75ab773-a0c3-472b-85de-ab1633cdd9d6">

**Comment by kwin** [comment]
This may happen if the execution takes too long. Please look at the error.log of the affected AEM instance which contains also the output from AEM during the build image phase. That should contain further hints. Compare with https://github.com/Netcentric/accesscontroltool/issues/669.

**Reply by teropikala** [reply]
Looks like it doesn't run it when there are no changes to the configuration files. Another try with small change seemed to work just fine. 

I'll check this again in the client environment where I initially saw the problem.

**Comment by ghenzler** [comment]
There is an explicit check if the config has changed and the tool is on purpose *not* executed, if the config has not changed (compare https://github.com/Netcentric/accesscontroltool/blob/d1fae6cff08212e82f16e2f74a967848a956549f/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImpl.java#L264) This is because immutable content can become very large.
